### ISSUE TITLE
Upgrade fails with "generate-keys" does not exist

### ISSUE THREAD

**Issue by ilvez** [firefly-iii/firefly-iii#2384]
I am running Firefly III version 4.7.17.6.

**Description**
I want to upgrade to 4.8.0. I'm using **Created using composer “create-project”** in Upgrade documentation. I get `The command "generate-keys" does not exist.` error running `php artisan firefly-iii:upgrade-database`.

**Extra info**
Before running these commands I only followed routine backup.

```
www-data@merkuur:~$ composer create-project grumpydictator/firefly-iii --no-dev --prefer-dist firefly-iii-updated 4.8.0
Installing grumpydictator/firefly-iii (4.8.0)
  - Installing grumpydictator/firefly-iii (4.8.0): Downloading (100%)         
Created project in firefly-iii-updated
> @php -r "file_exists('.env') || copy('.env.example', '.env');"
> @php -r "if (!(getenv('DYNO'))===false){file_exists('.env') || copy('.deploy/heroku/.env.heroku', '.env');copy('.deploy/heroku/.locales', '.locales');}"
Loading composer repositories with package information
Installing dependencies from lock file
Package operations: 85 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-ctype (v1.12.0): Downloading (100%)         
  - Installing phpoption/phpoption (1.5.0): Loading from cache
  - Installing vlucas/phpdotenv (v3.4.0): Downloading (100%)         
  - Installing symfony/css-selector (v4.3.3): Downloading (100%)         
  - Installing tijsverkoyen/css-to-inline-styles (2.2.1): Loading from cache
  - Installing symfony/polyfill-php72 (v1.12.0): Downloading (100%)         
  - Installing symfony/polyfill-mbstring (v1.12.0): Downloading (100%)         
  - Installing symfony/var-dumper (v4.3.3): Downloading (100%)         
  - Installing symfony/routing (v4.3.3): Downloading (100%)         
  - Installing symfony/process (v4.3.3): Downloading (100%)         
  - Installing symfony/polyfill-php73 (v1.12.0): Downloading (100%)         
  - Installing symfony/polyfill-intl-idn (v1.12.0): Downloading (100%)         
  - Installing symfony/mime (v4.3.3): Downloading (100%)         
  - Installing symfony/http-foundation (v4.3.3): Downloading (100%)         
  - Installing symfony/event-dispatcher-contracts (v1.1.5): Downloading (100%)         
  - Installing symfony/event-dispatcher (v4.3.3): Downloading (100%)         
  - Installing psr/log (1.1.0): Loading from cache
  - Installing symfony/debug (v4.3.3): Downloading (100%)         
  - Installing symfony/http-kernel (v4.3.3): Downloading (100%)         
  - Installing symfony/finder (v4.3.3): Downloading (100%)         
  - Installing psr/container (1.0.0): Loading from cache
  - Installing symfony/service-contracts (v1.1.5): Downloading (100%)         
  - Installing symfony/console (v4.3.3): Downloading (100%)         
  - Installing symfony/polyfill-iconv (v1.12.0): Downloading (100%)         
  - Installing doctrine/lexer (1.1.0): Downloading (100%)         
  - Installing egulias/email-validator (2.1.10): Downloading (100%)         
  - Installing swiftmailer/swiftmailer (v6.2.1): Downloading (100%)         
  - Installing paragonie/random_compat (v9.99.99): Loading from cache
  - Installing ramsey/uuid (3.8.0): Loading from cache
  - Installing psr/simple-cache (1.0.1): Loading from cache
  - Installing opis/closure (3.3.1): Downloading (100%)         
  - Installing symfony/translation-contracts (v1.1.5): Downloading (100%)         
  - Installing symfony/translation (v4.3.3): Downloading (100%)         
  - Installing nesbot/carbon (2.22.3): Downloading (100%)         
  - Installing monolog/monolog (1.24.0): Loading from cache
  - Installing league/flysystem (1.0.53): Downloading (100%)         
  - Installing erusev/parsedown (1.7.3): Downloading (100%)         
  - Installing dragonmantank/cron-expression (v2.3.0): Downloading (100%)         
  - Installing doctrine/inflector (v1.3.0): Loading from cache
  - Installing laravel/framework (v5.8.31): Downloading (100%)         
  - Installing tightenco/collect (v5.8.31): Downloading (100%)         
  - Installing adldap2/adldap2 (v10.0.11): Downloading (100%)         
  - Installing adldap2/adldap2-laravel (v6.0.7): Downloading (100%)         
  - Installing ralouphie/getallheaders (3.0.3): Downloading (100%)         
  - Installing psr/http-message (1.0.1): Loading from cache
  - Installing guzzlehttp/psr7 (1.6.1): Downloading (100%)         
  - Installing guzzlehttp/promises (v1.3.1): Loading from cache
  - Installing guzzlehttp/guzzle (6.3.3): Loading from cache
  - Installing bunq/sdk_php (dev-master be64573): Downloading (100%)         
  - Installing danhunsaker/laravel-flysystem-service (v1.2.2): Loading from cache
  - Installing danhunsaker/laravel-flysystem-others (v1.3.4): Loading from cache
  - Installing davejamesmiller/laravel-breadcrumbs (5.2.1): Loading from cache
  - Installing doctrine/event-manager (v1.0.0): Loading from cache
  - Installing doctrine/cache (v1.8.0): Loading from cache
  - Installing doctrine/dbal (v2.9.2): Loading from cache
  - Installing fideloper/proxy (4.2.0): Downloading (100%)         
  - Installing psr/http-factory (1.0.1): Downloading (100%)         
  - Installing zendframework/zend-diactoros (2.1.3): Downloading (100%)         
  - Installing symfony/psr-http-message-bridge (v1.2.0): Loading from cache
  - Installing phpseclib/phpseclib (2.0.21): Downloading (100%)         
  - Installing league/event (2.2.0): Loading from cache
  - Installing lcobucci/jwt (3.3.1): Downloading (100%)         
  - Installing defuse/php-encryption (v2.2.1): Loading from cache
  - Installing league/oauth2-server (7.4.0): Downloading (100%)         
  - Installing firebase/php-jwt (v5.0.0): Loading from cache
  - Installing laravel/passport (v7.3.5): Downloading (100%)         
  - Installing laravelcollective/html (v5.8.0): Loading from cache
  - Installing league/commonmark (1.0.0): Downloading (100%)         
  - Installing league/csv (9.3.0): Downloading (100%)         
  - Installing league/flysystem-replicate-adapter (1.0.1): Loading from cache
  - Installing league/flysystem-sftp (1.0.20): Downloading (100%)         
  - Installing league/fractal (0.18.0): Downloading (100%)         
  - Installing litipk/flysystem-fallback-adapter (0.1.3): Loading from cache
  - Installing mschindler83/fints-hbci-php (1.0.4): Loading from cache
  - Installing paragonie/constant_time_encoding (v2.2.3): Loading from cache
  - Installing symfony/polyfill-util (v1.12.0): Downloading (100%)         
  - Installing symfony/polyfill-php56 (v1.12.0): Downloading (100%)         
  - Installing pragmarx/google2fa (v5.0.0): Downloading (100%)         
  - Installing bacon/bacon-qr-code (1.0.3): Loading from cache
  - Installing pragmarx/google2fa-qrcode (v1.0.3): Downloading (100%)         
  - Installing pragmarx/google2fa-laravel (v1.0.1): Downloading (100%)         
  - Installing pragmarx/random (v0.2.2): Downloading (100%)         
  - Installing pragmarx/recovery (v0.1.0): Downloading (100%)         
  - Installing twig/twig (v1.42.2): Downloading (100%)         
  - Installing rcrowe/twigbridge (v0.9.12): Loading from cache
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan firefly:instructions install
+------------------------------------------------------------------------------+
|                                                                              |
| Thank you for installing Firefly III, v4.8.0!                                |
| This is a huge upgrade for Firefly III. Please expect bugs and errors, and   |
| bear with me as I fix them. I tested a lot of things but pretty sure I       |
| missed some. Thanks for understanding.                                       |
|                                                                              |
+------------------------------------------------------------------------------+
> @php artisan key:generate
Application key set successfully.
www-data@merkuur:~$ cp firefly-iii/.env firefly-iii-updated/.env
www-data@merkuur:~$ cp firefly-iii/storage/upload/* firefly-iii-updated/storage/upload/
cp: cannot stat 'firefly-iii/storage/upload/*': No such file or directory
www-data@merkuur:~$ cp firefly-iii/storage/export/* firefly-iii-updated/storage/export/
www-data@merkuur:~$ cd firefly-iii-updated
www-data@merkuur:~/firefly-iii-updated$ rm -rf bootstrap/cache/*
www-data@merkuur:~/firefly-iii-updated$ php artisan cache:clear
Application cache cleared!
www-data@merkuur:~/firefly-iii-updated$ php artisan migrate --seed
Migrating: 2019_03_22_183214_changes_for_v480
Migrated:  2019_03_22_183214_changes_for_v480 (0.03 seconds)
Seeding: AccountTypeSeeder
Seeding: TransactionCurrencySeeder
Seeding: TransactionTypeSeeder
Seeding: PermissionSeeder
Seeding: LinkTypeSeeder
Seeding: ConfigSeeder
Database seeding completed successfully.
www-data@merkuur:~/firefly-iii-updated$ php artisan firefly-iii:upgrade-database

                                               
  The command "generate-keys" does not exist.  
                                               

```

Thank you very much for big upgrade, was waiting patiently and now it's here!

**Comment by JC5** [comment]
Thanks :)

To fix this, please find [this file](https://github.com/firefly-iii/firefly-iii/blob/master/app/Console/Commands/Upgrade/UpgradeDatabase.php#L117) in your `firefly-iii-updated` directory.

Then change the content of line 117 to the following:

```php
Artisan::call('passport:install');
```

Then run `php artisan firefly-iii:upgrade-database` again.

A fix for this will be available next week in version 4.8.0.1

**Comment by ilvez** [comment]
Cheers! Thanks for quick response. Step completed successfully.
